File: src/Leeson7/Homework7/Java/Author.java
package Leeson7.Homework7.Java;

public class Author {

    private String name;
    private int year;

    public Author(String name, int year) {
        this.name = name;
        this.year = year;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || this.getClass() != obj.getClass()) return false;

        return this.name.equals(((Author) obj).name);
    }
}

File: src/Leeson7/Homework7/Kotlin/Author.kt
package Leeson7.Homework7.Kotlin

import Leeson7.Homework7.Java.Author

class Author(val name : String, private val year: Int) {

    override fun equals(otherAuthor: Any?): Boolean {
        if (otherAuthor == null || otherAuthor !is Author) return false
        if (this::class === otherAuthor) return true

        return this.name.equals(otherAuthor)
    }

    override fun hashCode(): Int {
        return name.hashCode() + year
    }
}